### ISSUE TITLE
postfixadmin doesn't work right with HTTP/3

### ISSUE THREAD

**Issue by elyograg** [postfixadmin/postfixadmin#614]
I updated my webserver to include HTTP/3 yesterday.  Users connect to haproxy, which handles all the encryption and things like http/3. proxying connections to Apache on the same machine using h2c.

Today I found that when I do certain things with postfixadmin, wihch includes logging in, that I get a 400 response if I   am connecting with HTTP/3.  I configured haproxy to not send the alt-svc header if the URL path begins with /postfixadmin and now everything is fine, the connection uses HTTP/2 as it always has before.  Other PHP apps work correctly via HTTP/3, like roundcube.

I've got postfixadmin 3.3.8, Apache 2.4.41-4ubuntu3.10, and haproxy 2.6.2-dev5.  I was using firefox to conen=n

**Comment by elyograg** [comment]
Maybe I spoke too soon.  roundcube does show the same 400 error at login.  I saw it work before, so I am confused.

Here's the haproxy log for a 400 error on login:

`Apr 11 07:19:14 - haproxy[1526259] 73.131.246.74:50262 [11/Apr/2022:07:19:14.484] web be-apache-81/bilbo 1/0/0/-1/0 400 206 - - CH-- 2/2/0/0/0 0/0 {webmail.elyograg.org} "POST /postfixadmin/login.php HTTP/3.0"`

That indicates that the 400 came from Apache.  Apache doesn't have anything.  The error log is empty and the access log at that timestamp only shows the "HEAD /" requests that haproxy makes every ten seconds for a health check.

**Comment by DavidGoodwin** [comment]
Well, you should be able to issue curl requests on the backend itself, to confirm that e.g. Apache is behaving properly.

You'll need to be running Apache with the http2 module enabled if you want to have a http2 compatible backend. 

That'll require PHP runs through php-fpm.

Does everything work if you allow haproxy to talk 'boring' http 1.1?

**Comment by elyograg** [comment]
The problem's not in PHP.  I can demonstrate it.

Open the following link  in Chrome.  It must be Chrome.

https://raspi2.elyograg.org/form.html

Hold shift while clicking the reload button. That will load the page via HTTP/3.  This is a simple html form that issues a POST request.  I grabbed it from the internet.  When you click the button to send the greeting, you'll get a 400 error.

**Comment by DavidGoodwin** [comment]
I'm assuming the problem is outside of Postfixadmin and closing this.

**Comment by elyograg** [comment]
It was a general problem doing POST requests via HTTP/3 to haproxy, which then speaks HTTP/2 to Apache.  Any POST request would return a 400 error.  That led to a bugfix for haproxy in ttheir QUIC/HTTP3 implementation.  Still has some problems -- I have a paste website at https://paste.elyograg.org ... this fails to work over HTTP/3 for large content (43000 lines in the data I was trying to upload).  HTTP/3 support in haproxy is still experimental.

Long story short:  postfixamin is not the problem. :)
